### PR TITLE
[tests] Use `probes.json` instead of `vercel.json` for e2e tests

### PR DESCRIPTION
**Pull request by TooTallNate**
Makes it easier to test fixtures directly via CLI deployments.

Coincidentally, this also revealed that auto-detection of Remix apps wasn't working correctly if there's a `remix.config.mjs` file. So `@vercel/frameworks` is updated to account for that case as well.

### CODE REVIEW

File: packages/frameworks/src/frameworks.ts [hunk 1]
```diff
@@ -202,12 +202,13 @@ export const frameworks = [
...
-      every: [
-        // Intentionally does not detect a package name
-        // https://github.com/vercel/vercel/pull/7761
+      some: [
         {
           path: 'remix.config.js',
         },
+        {
+          path: 'remix.config.mjs',
+        },
```
Comment (styfle) [inline packages/frameworks/src/frameworks.ts]: Let's separate this change into a different PR because the changes won't take affect until its updated in `vercel/api`

Comment (TooTallNate) [reply packages/frameworks/src/frameworks.ts]: Sounds good.

Comment (TooTallNate) [reply packages/frameworks/src/frameworks.ts]: https://github.com/vercel/vercel/pull/9558

**Review by styfle — APPROVED** [review]
Thanks! One of these days we need to run a script to update all of the fixtures in the repo 😄
